modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AlterTableAlterColumnCommand.java

```diff
@@ -153,7 +152,8 @@ private void validateColumnChange(CatalogTableColumnDescriptor origin) {
             }
         }
 
-        if (precision != null && precision != origin.precision() && origin.type() != ColumnType.DECIMAL) {
+        if (precision != null && precision != origin.precision() && (origin.type() != ColumnType.DECIMAL
+                || origin.type().precScale() == PrecisionScale.NO_NO)) {
```

Comment:
This looks redundant, because `ColumnType.DECIMAL.precScale` equals to `PrecScale.YES_YES`